### PR TITLE
CO-1753: replace age mapping for child by computed field

### PR DESCRIPTION
**Pull request by samuel-fringeli**

### CODE REVIEW

**Comment by ecino** [comment]
Please check LINT errors

**Comment by samuel-fringeli** [comment]
Pylint errors are corrected, but it stays flake8 errors that come from hr_attendance_extra_hours. However, there is no flake8 errors related to my commit.

File: child_compassion/models/global_child.py [hunk 1]
```diff
@@ -97,6 +98,14 @@ def get_child_vals(self):
         del vals['id']
         return vals
 
+    @api.multi
+    def _compute_age(self):
+        for child in self:
+            today = date.today()
```
Comment (ecino) [inline child_compassion/models/global_child.py]: Put this before the loop to improve optimization

File: child_compassion/models/global_child.py [hunk 2]
```diff
@@ -97,6 +98,14 @@ def get_child_vals(self):
...
+            born = datetime.strptime(child.birthdate, '%Y-%m-%d').date()
```
Comment (ecino) [inline child_compassion/models/global_child.py]: Better : `born = fields.Date.from_string(child.birthdate)`

File: partner_communication/wizards/change_text_wizard.py [hunk 3]
```diff
@@ -29,11 +28,12 @@ def _default_text(self):
         lang = list(set(communications.mapped('partner_id.lang')))
         if len(config) != 1:
             raise UserError(
-                _("You can only update text on one communication type at "
+                ("You can only update text on one communication type at "
```
Comment (ecino) [inline partner_communication/wizards/change_text_wizard.py]: No! Put back that `_` and put it back from the imports. This is a translation function that translates the error strings to the user language.

File: partner_communication/wizards/change_text_wizard.py [hunk 4]
```diff
@@ -29,11 +28,12 @@ def _default_text(self):
...
                   "time."))
         if len(lang) != 1:
             raise UserError(
-                _("Please update only one language at a time."))
+                ("Please update only one language at a time."))
```
Comment (ecino) [inline partner_communication/wizards/change_text_wizard.py]: Put back translation

File: partner_communication/wizards/change_text_wizard.py [hunk 5]
```diff
@@ -48,10 +48,13 @@ def update(self):
         template = config.email_template_id.with_context(lang=lang)
         if len(config) != 1:
             raise UserError(
-                _("You can only update text on one communication type at "
+                ("You can only update text on one communication type at "
```
Comment (ecino) [inline partner_communication/wizards/change_text_wizard.py]: Put back translation

**Review by ecino — CHANGES_REQUESTED** [review]

**Comment by ecino** [comment]
These are your flake8 errors:
```
child_compassion/models/global_child.py:107:21: E126 continuation line over-indented for hanging indent
partner_communication/wizards/change_text_wizard.py:15:1: E302 expected 2 blank lines, found 1
partner_communication/wizards/change_text_wizard.py:18:80: E501 line too long (84 > 79 characters)
partner_communication/wizards/change_text_wizard.py:32:19: E127 continuation line over-indented for visual indent
partner_communication/wizards/change_text_wizard.py:52:19: E127 continuation line over-indented for visual indent
```

**Comment by samuel-fringeli** [comment]
Voilà, j'ai tout corrigé

**Review by ecino — APPROVED** [review]

**Pull request merged**